### PR TITLE
Add ledger operation validation.

### PR DESCRIPTION
**Pull request by gannan08**

### CODE REVIEW

File: lib/LedgerNodeOperations.js [hunk 1]
```diff
@@ -46,9 +48,41 @@ module.exports = class LedgerNodeOperations {
...
+  _validateContext(context);
+  const validation = validate('webledger-validator.operation', operation);
+  if(!validation.valid) {
+    throw validation.error;
+  }
+}
+
+function _validateContext(context) {
+  const wlContextUrl = constants.WEB_LEDGER_CONTEXT_V1_URL;
+  if(!context || !(typeof context === 'string' || Array.isArray(context))) {
```
Comment (mattcollier) [inline lib/LedgerNodeOperations.js]: It appears to me that this bit is handled by the JSON schema validation.  Is that right?

File: package.json [hunk 2]
```diff
@@ -17,6 +17,8 @@
   },
   "homepage": "https://github.com/digitalbazaar/bedrock-ledger-node",
   "dependencies": {
+    "bedrock-ledger-context": "^1.1.1",
```
Comment (mattcollier) [inline package.json]: these should be peer deps here right?  And real deps in the test framework.

File: lib/LedgerNodeOperations.js (hunk 1 above)
Comment (gannan08) [reply lib/LedgerNodeOperations.js]: Correct

Comment (dlongley) [reply lib/LedgerNodeOperations.js]: What JSON schema validator handles this? We need to make sure it doesn't allow a different order on the context(s).

Comment (mattcollier) [reply lib/LedgerNodeOperations.js]: @dlongley the part that you're referring to is still in there.  The part we didn't need has already been removed and that's simple existence check and type checking which is already handled by JSON schema.

Comment (dlongley) [reply lib/LedgerNodeOperations.js]: Ok.

**Comment by mattcollier** [comment]
@dlongley continuity patch is in, waiting on your approval here.

File: lib/LedgerNodeOperations.js [hunk 3]
```diff
@@ -31,7 +33,7 @@ module.exports = class LedgerNodeOperations {
    */
   async add({operation}) {
     if(!operation) {
-      throw new TypeError('`operation` is required.');
+      throw new BedrockError('`operation` is required.', 'TypeError');
```
Comment (dlongley) [inline lib/LedgerNodeOperations.js]: Why `BedrockError` instead of `TypeError`?

File: lib/LedgerNodeOperations.js [hunk 4]
```diff
@@ -46,9 +48,37 @@ module.exports = class LedgerNodeOperations {
    * @return a Promise that resolves once the operation completes.
    */
   async validate(operation) {
+    if(!(operation && typeof operation === 'object')) {
+      throw new BedrockError('Operation must be an object.', 'TypeError');
```
Comment (dlongley) [inline lib/LedgerNodeOperations.js]: Same question here ... our pattern is to use `TypeError` for this sort of thing.

File: lib/LedgerNodeOperations.js [hunk 5]
```diff
@@ -46,9 +48,37 @@ module.exports = class LedgerNodeOperations {
...
+  if(!validation.valid) {
+    throw validation.error;
+  }
+}
+
+function _validateContext(context) {
+  const wlContextUrl = constants.WEB_LEDGER_CONTEXT_V1_URL;
+  if(typeof context === 'string' && context !== wlContextUrl) {
+    throw new BedrockError(`Operation context must be "${wlContextUrl}"`,
+      'SyntaxError', {context});
```
Comment (dlongley) [inline lib/LedgerNodeOperations.js]: Style nit: If you drop any params to a function to a new line, always drop the first one too.

**Pull request closed without merging**